src/main/java/org/mule/tools/devkit/sonar/checks/structure/GitIgnoreExistsCheck.java

```diff
@@ -0,0 +1,80 @@
+package org.mule.tools.devkit.sonar.checks.structure;
+
+import com.google.common.base.Predicates;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.project.MavenProject;
+import org.mule.tools.devkit.sonar.checks.ConnectorIssue;
+import org.sonar.api.batch.fs.FileSystem;
+import org.apache.commons.io.FileUtils;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Rule(key = GitIgnoreExistsCheck.KEY, name = ".gitignore should be present", description = "There should exist a .gitignore file in root folder.", priority = Priority.CRITICAL, tags = { "connector-certification" })
+public class GitIgnoreExistsCheck  extends ExistingResourceCheck {
+
+
+    public static final String KEY = "gitignore-exist";
+    private static final String PATH = ".gitignore";
+    private static final ImmutableList<String> REQUIRED_GITIGNORE_FIELDS = ImmutableList.of("*.class", "*.jar", "*.war","target/",".classpath", ".settings/", ".project", ".factorypath",".idea/", "*.iml", "*.ipr", "*.iws", "bin/", ".DS_Store","automation-credentials.properties");
+
+
+    private static FileSystem fileSystem;
+
+    public GitIgnoreExistsCheck(FileSystem fileSystem) {
+        super(fileSystem);
```

Comment:
It doesn't make much sense to extend `ExistingResourceCheck`, given the only method it provides is overwritten completely. Let's remove the inheritance step.